src/analytics/internal/send-to-server/send-to-server.js

```diff
@@ -0,0 +1,129 @@
+import { idleManager } from 'src/util/idle'
+import { SHOULD_TRACK_STORAGE_KEY as SHOULD_TRACK } from 'src/options/privacy/constants'
+import { USER_ID, generateTokenIfNot } from 'src/util/generate-token'
+import { installTimeStorageKey } from 'src/imports/background'
+
+class SendToServer {
+    static API_PATH = '/event'
+    static JSON_HEADER = {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+    }
+
+    /**
+     * Pool of requests that have been tracked, which will be periodically cleared and sent off in bulk.
+     */
+    _pool = new Set()
+
+    /**
+     * @param {Object} args
+     * @param {string} args.url Full URL of the analytics server host.
+     */
+    constructor({ url }) {
+        this._host = url + SendToServer.API_PATH
+
+        // Schedule sending of network req when user is idle or locks screen
+        idleManager.scheduleIdleCbs({
+            onIdle: this._sendBulkReq,
+            onLocked: this._sendBulkReq,
+        })
+    }
+
+    /**
+     *
+     * @param {params} params if params is there, then send request directly without wait, otherwise send pool request
+     */
+    static _serializePoolReqs(userId, events) {
+        const data = {
+            id: userId,
+            data: [...events],
+        }
+
+        return JSON.stringify(data)
+    }
+
+    _poolReq = params => this._pool.add(params)
+
+    /**
+     * Send a request to the Redash HTTP Tracking API. Takes care of calculating all default
+     * or easily derivable params, merging them with input `params`.
+     *
+     * @param {any} params
+     * @return {Promise<Response>}
+     */
+    _sendReq = async event => {
+        const userId = await this.userId()
+
+        await fetch(this._host, {
```

Comment:
This should probably have a `return` so the `Response` can be handled somewhere.